packages/hint-leading-dot-classlist/src/hint.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview Inform users about classList.add or classList.remove with leading '.' in the argument pitfall
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint } from 'hint/dist/src/lib/types';
+import { debug as d } from '@hint/utils-debug';
+import { Severity } from '@hint/utils-types';
+import { ScriptEvents } from '@hint/parser-javascript';
+
+import meta from './meta';
+import { getMessage } from './i18n.import';
+
+const debug: debug.IDebugger = d(__filename);
+
+export default class LeadingDotClasslist implements IHint {
+
+    public static readonly meta = meta;
+
+    public constructor(context: HintContext<ScriptEvents>) {
+
+        /*
+         * Check to see if any call expression in the AST has a callee with
+         * a 'classList.add' or 'classList.remove' property. If yes, check if
+         * that call expression's first argument contains a leading '.'. In
+         * that case, report a warning as the leading '.' may lead to an
+         * undesireable pitfall
+         */
+        context.on('parse::end::javascript', ({ ast, element, resource, sourceCode, walk }) => {
+
+            debug('Validating hint classlist-add-remove-pitfall');
+
+            walk.simple(ast, {
+                CallExpression(node) {
+                    if (!('property' in node.callee && 'name' in node.callee.property)) {
+                        return;
+                    }
+
+                    if ((node.callee.property.name !== 'add' && node.callee.property.name !== 'remove') || (node.callee as any).object?.property?.name !== 'classList') {
+                        return;
+                    }
```

Comment:
Hey @ruppysuppy, sorry for my slow response. I dug into the type issue a bit more and realized the types are correct but we need a couple of additional checks to narrow them down. This is because there are multiple possible types for `callee`, `object`, etc. which don't all have the same properties.
```suggestion
                    if (node.callee.property.name !== 'add' && node.callee.property.name !== 'remove') {
                        return;
                    }
                    
                    if (!('object' in node.callee && 'property' in node.callee.object && 'name' in node.callee.object.property && node.callee.object.property.name === 'classList')) {
                        return;
                    }
```